File: Python/Problems/Problems/problem3222.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem3222(Problem):
    def test(self) -> bool:
        x = 2
        y = 7

        expected = "Alice"

        result = self.winningPlayer(x, y)

        print(result)

        return expected == result

    def winningPlayer(self, x: int, y: int) -> str:
        yTurns = y // 4
        return "Bob" if min(x, yTurns) % 2 == 0 else "Alice"

File: C++/Problems/Problems/problem_3222.cpp
#include "problem_3222.h"

bool problem_3222::test() {
    int x = 2;
    int y = 7;

    std::string expected = "Alice";

    auto result = winningPlayer(x, y);

    std::cout << result << std::endl;

    return expected == result;
}

std::string problem_3222::winningPlayer(int x, int y) {
    int yTurns = y / 4;
    return std::min(x, yTurns) % 2 == 0 ? "Bob" : "Alice";
}
